lib/remote/objectqueryhandler.cpp

```diff
@@ -257,6 +257,15 @@ bool ObjectQueryHandler::HandleRequest(
 			if (!joinedObj)
 				continue;
 
+			String permission = "objects/query/" + joinedObj->GetReflectionType()->GetName();
+
+			if (!FilterUtility::HasPermission(user, permission)) {
+				// The API user isn't authorized to access this relation, so just ignore it because there
+				// is no reason to raise an exception here since the actual object requested by the client
+				// has been found!!
+				continue;
+			}
+
```

Comment:
OK, in case of insufficient permission you skip it... but otherwise you **unconditionally** include it, right? This is a also bad idea. Figure out and explain in your own words why.

Reply:
Have to evaluate the permission filters as well!

Reply:
This comment should be resolved, shouldn't it?